src/main/java/com/sngular/kloadgen/sampler/KafkaProducerSampler.java

```diff
@@ -100,23 +100,29 @@ public void setupTest(final JavaSamplerContext context) {
       props2.putIfAbsent(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, AvroKafkaSerializer.class.getName());
 
       // Configure Service Registry location
-      props2.putIfAbsent(SerdeConfig.REGISTRY_URL, "http://localhost:8080");
+      //      props2.putIfAbsent(SchemaRegistryKeyHelper.SCHEMA_REGISTRY_URL, props.get(SchemaRegistryKeyHelper.SCHEMA_REGISTRY_URL));
+      //      props2.putIfAbsent(PropsKeysHelper.VALUE_SUBJECT_NAME_STRATEGY, props.get(PropsKeysHelper.VALUE_SUBJECT_NAME_STRATEGY));
 
       // Register the artifact if not found in the registry.
       // props2.putIfAbsent(SerdeConfig.AUTO_REGISTER_ARTIFACT, Boolean.TRUE);
 
-      props2.putIfAbsent("apicurio.registry.artifact-resolver-strategy", props.get(ProducerKeysHelper.VALUE_NAME_STRATEGY));
+      //      props2.putIfAbsent(SchemaResolverConfig.ARTIFACT_RESOLVER_STRATEGY, props.get(ProducerKeysHelper.VALUE_NAME_STRATEGY));
 
-      producer = new KafkaProducer<>(props2);
+      producer = new KafkaProducer<>(props);
     } catch (final KafkaException ex) {
       getNewLogger().error(ex.getMessage(), ex);
     }
   }
 
   private Properties properties(final JavaSamplerContext context) {
     final var commonProps = SamplerUtil.setupCommonProperties(context);
-    if (Objects.nonNull(context.getParameter(ProducerKeysHelper.VALUE_NAME_STRATEGY))) {
-      commonProps.put(ProducerKeysHelper.VALUE_NAME_STRATEGY, context.getParameter(ProducerKeysHelper.VALUE_NAME_STRATEGY));
+
+    String artifactResolverStrategy = context.getParameter(ProducerKeysHelper.VALUE_NAME_STRATEGY);
+    if (Objects.nonNull(artifactResolverStrategy)) {
+      commonProps.put(ProducerKeysHelper.VALUE_NAME_STRATEGY, artifactResolverStrategy);
+    }
+    if (Objects.nonNull(context.getParameter(SchemaResolverConfig.ARTIFACT_RESOLVER_STRATEGY))) {
```

Comment:
Please, review this `if` because without context it seems strange.
If a parameter is filled, we will add a paramter to another list with a different value (that could be `null` according to the previous `if`, and it could has been included in the same list with a different key).

Reply:
I agree. It seems strange but the value key used by SerDesPropertyEditors to storage serializer value selected by user is the same than used internally by Confluente libraries. However, Apicurio uses 'apicurio.registry.artifact-resolver-strategy' as key value to manage its serializer values.